commons/com.b2international.index/src/com/b2international/index/es/EsDocumentSearcher.java

```diff
@@ -233,7 +230,7 @@ public EsDocumentSearcher(EsIndexAdmin admin, ObjectMapper mapper) {
 	private <T> boolean applySourceFiltering(List<String> fields, boolean isDocIdOnly, final DocumentMapping mapping, final SearchSourceBuilder reqSource) {
 		// No specific fields requested? Use _source to retrieve all of them
 		if (fields.isEmpty()) {
-			reqSource.fetchSource(null, EXCLUDED_SOURCE_FIELDS);
```

Comment:
The 0-arg version of `fetchSource` returns a `FetchSourceContext` which holds the currently configured values, but does not modify any behavior: https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourceContext.java

If you'd like to control source filtering, use the single boolean form, but retrieving the source document is already the default: https://www.elastic.co/guide/en/elasticsearch/client/java-rest/master/java-rest-high-search.html#_source_filtering